### PR TITLE
yang models for device_metadata, device_neighbor, flex_counters, crm, versions. Processing code and tests.

### PR DESCRIPTION
**Pull request by praveen-li**
**- What I did**
yang models for device_metadata, device_neighbor, flex_counters, crm, versions. Processing code and tests.

**- How I did it**
Support container(s), list(s) and leaves inside a container.

**- How to verify it**
Added 2 new tests in sonic_yang_models and 1 new test in sonic_yang_mgmt.
```
116 INFO:YANG-TEST:
117 ------------------- Test 13: CRM BREAKOUT CFG FLEX COUNTER TABLE.---------------------
118 INFO:YANG-TEST:CRM BREAKOUT CFG FLEX COUNTER TABLE. Passed

134 INFO:YANG-TEST:
135 ------------------- Test 17: DEVICE_METADATA DEVICE_NEIGHBOR VERSION TABLE.---------------------
136 INFO:YANG-TEST:DEVICE_METADATA DEVICE_NEIGHBOR VERSION TABLE. Passed

pchaudha@server05:/home/pchaudha/srcCode/dpb_repo$ ls -l target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl
-rw-r--r-- 1 pchaudha pchaudha 18591 Jul 14 10:41 target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl
pchaudha@server05:/home/pchaudha/srcCode/dpb_repo$ date
Tue Jul 14 10:43:04 PDT 2020

pchaudha@server05:/home/pchaudha/srcCode/dpb_repo$ ls -l target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl
-rw-r--r-- 1 pchaudha pchaudha 19963 Jul 14 11:09 target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl
S
pchaudha@server05:/home/pchaudha/srcCode/dpb_repo$ ls -l target/python-debs/python-sonic-utilities_1.2-1_all.deb
-rw-r--r-- 1 pchaudha pchaudha 153222 Jul 14 11:11 target/python-debs/python-sonic-utilities_1.2-1_all.deb
```

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 1]
```diff
@@ -297,41 +297,86 @@ def xlateList(self, model, yang, config, table):
...
-                if len(yang[modelList['@name']]) == 0:
-                    del yang[modelList['@name']]
+                self.xlateListInContainer(modelList, yang, configC, table)
 
+        # Handle container(s) in container
+        ccontainer = model.get('container')
+        # If single list exists in container,
+        if ccontainer and isinstance(ccontainer, dict) and bool(configC):
+            self.xlateContainerInContainer(ccontainer, yang, configC, table)
+        # If multi-list exists in container,
```
Comment (praveen-li) [inline src/sonic-yang-mgmt/sonic_yang_ext.py]: /s/multi-list/multi-container/

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 2]
```diff
@@ -446,23 +491,60 @@ def revXlateList(self, model, yang, config, table):
...
+        # IF container has lists
+        elif isinstance(clist, list):
+            for modelList in clist:
+                self.revXlateListInContainer(modelList, yang, config, table)
+
+        ccontainer = model.get('container')
+        # IF container has only one inner container
+        if isinstance(ccontainer, dict):
+            self.revXlateContainerInContainer(ccontainer, yang, config, table)
+        # IF container has only many inner container
```
Comment (vasant17) [inline src/sonic-yang-mgmt/sonic_yang_ext.py]: delete only word?

File: src/sonic-yang-models/yang-models/sonic-crm.yang [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+        container CRM {
+
+            description "CRM part of config_db.json";
+
+            container Config {
+
+                /* typedef specific to CRM */
+                typedef threshold {
+                    type uint16;
```
Comment (li-pingmao) [inline src/sonic-yang-models/yang-models/sonic-crm.yang:29]: threshold range?

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 4]
```diff
@@ -446,23 +491,60 @@ def revXlateList(self, model, yang, config, table):
...
+            for modelContainer in ccontainer:
+                self.revXlateContainerInContainer(modelContainer, yang, config, table)
+
+        ## Handle other leaves in container,
+        leafDict = self.createLeafDict(model)
+        for vKey in yang:
+            #vkey must be a leaf\leaf-list\choice in container
+            if leafDict.get(vKey):
+                self.sysLog(syslog.LOG_DEBUG, "revXlateContainer vkey {}".format(vKey))
+                config[vKey] = self.revFindYangTypedValue(vKey, yang[vKey], leafDict)
```
Comment (li-pingmao) [inline src/sonic-yang-mgmt/sonic_yang_ext.py:548]: Supports one level of containers/lists in container?   Any need to support more levels of containers/lists in containers?

File: src/sonic-yang-mgmt/sonic_yang_ext.py (hunk 1 above)
Comment (praveen-li) [reply src/sonic-yang-mgmt/sonic_yang_ext.py]: done

File: src/sonic-yang-mgmt/sonic_yang_ext.py (hunk 2 above)
Comment (praveen-li) [reply src/sonic-yang-mgmt/sonic_yang_ext.py]: done

File: src/sonic-yang-models/yang-models/sonic-crm.yang (hunk 3 above)
Comment (praveen-li) [reply src/sonic-yang-models/yang-models/sonic-crm.yang:29]: No constraints are defined right now, except it is percentage. That why I kept below to handle the percentage part.

                    must "(((current()='PERCENTAGE' or current()='percentage') and
                        ../acl_counter_high_threshold<100 and
                        ../acl_counter_low_threshold<100) or
                        (current()!='PERCENTAGE' and current()!='PERCENTAGE'))";
                    type stypes:crm_threshold_type;

File: src/sonic-yang-mgmt/sonic_yang_ext.py (hunk 4 above)
Comment (praveen-li) [reply src/sonic-yang-mgmt/sonic_yang_ext.py:548]: Multiple lists case exists of INTERFACE and VLAN_INTERFACE Table after the VRF feature. 

Multiple containers case does not exist as of now, but I thought to support it, as in future, the conversion will fail if someone will write yang models with >=2 containers in another container.

**Review by li-pingmao — APPROVED** [review]

**Pull request merged**